src/day11.rs

```diff
@@ -1,25 +1,375 @@
+        }
+    );
+    let max = *max_power.lock().expect("Failed to get max power mutex to report result");
+
+    (max.1, max.2)
+}
+
+fn largest_nxn(grid: &Vec<Vec<i32>>) -> (i32, i32,i32) {
+    let arc_grid = Arc::new(grid);
+    let mut max_power = Mutex::new((0, 0, 0, 0)); // 0th is power, 1st is x, 2nd is y (should go write a struct really..), 3rd is size
+    rayon::scope(|s|
+        for (x_idx, col) in grid.iter().enumerate() {
+            let arc_grid = *arc_grid;
+            let max_power = &max_power;
+            s.spawn(move|s| {
+            for (y_idx, _) in col.iter().enumerate() {            
+                let mut max_local_power = 0;
+                let mut max_n = 0;
+                find_max_power_with_size(&mut max_local_power, &mut max_n, arc_grid, x_idx, y_idx);
+    
+                if max_local_power > max_power.lock().expect("Failed to get max power mutex for comparison").0 {
+                    *max_power.lock().expect("Failed to get max power mutex to change value") = (max_local_power, (x_idx+1) as i32, (y_idx+1) as i32, max_n);
+                    // println!("Updating max_value {}, {}, {}", max_local_power, (x_idx+1) as i32, (y_idx+1) as i32);
+                }
+            }    
+            // println!("Max power: {:?}", *max_power.lock().expect("Failed to get max power mutex to report result")); 
+        });
+    });
+    
+    let max = *max_power.lock().expect("Failed to get max power mutex to report result");
+    // println!("max power:{}", max.0);
+    (max.1, max.2, max.3 as i32)
+}
+
+fn add_to_grid(running_total: &mut i32, arc_grid: &Vec<Vec<i32>>, x_idx: usize, y_idx: usize, n: usize) {
+    // if n==16 && y_idx+1==269 && x_idx+1 == 90 {
+        // println!("Corner: {},", arc_grid[x_idx+n][y_idx+n]);
+        // println!("Vertical:");
+    // }
+    *running_total += arc_grid[x_idx+n][y_idx+n]; // corner
+    for y_idx2 in y_idx..(y_idx+n-1) {
+        // if n==16 && y_idx+1==269 && x_idx+1 == 90 {
+            // print!("{},", arc_grid[x_idx+n][y_idx2]);
+        // }
+        *running_total += arc_grid[x_idx+n][y_idx2]; // vertical strip
+    }
+    // if n==16 && y_idx+1==269 && x_idx+1 == 90 {
+        // println!("\nHorizontal:");
+    // }
+    for x_idx2 in x_idx..(x_idx+n-1) {
+        // if n==16 && y_idx+1==269 && x_idx+1 == 90 {
+        //     print!("{},", arc_grid[x_idx2][y_idx+n]);
+        // }
+        *running_total += arc_grid[x_idx2][y_idx+n]; // horizontal strip
+    }
+    // if n==16 && y_idx+1==269 && x_idx+1 == 90 {
+    // println!("");
+    // }
+    // if x_idx+1 == 90 && (y_idx+1 == 269 || y_idx+1 == 265){
+    //     println!("x:{},y:{}calculating {}x{} grid, total is now {}", x_idx+1, y_idx+1, n, n, running_total);
+    // }
+}
+
+fn find_max_power_with_size(max_local_power: &mut i32, max_n: &mut i32, arc_grid: &Vec<Vec<i32>>, x_idx: usize, y_idx: usize) {
+    let mut running_total = 0;
+    for n in 1..301 {
+        if x_idx + n < 300 && y_idx + n < 300 {
```

Comment:
Not a big deal since the checks will be very quick, but it's a bit wasteful to keep incrementing n after the inner check fails  since both the contained if tests will then fail for every other value of n.  You could rewrite with the upper bound pre-calculated, or with the opposite if test providing a break statement to skip the unnecessary checks.